planet/clients/orders.py

```diff
@@ -302,68 +302,51 @@ async def aggregated_order_stats(self) -> dict:
-                             filename: str = None,
-                             directory: str = None,
+    async def download_order(self,
+                             order: Order,
+                             directory: Path,
                              overwrite: bool = False,
-                             progress_bar: bool = True) -> str:
-        """Download ordered asset.
+                             callback=None,
+                             callback_args: tuple = None,
+                             callback_kwargs: dict = None) -> None:
+        """Download all assets in a finished order.
 
         Parameters:
-            location: Download location url including download token.
-            filename: Custom name to assign to downloaded file.
-            directory: Write to given directory instead of current directory.
+            order: a finished Order.
+            directory: a download directory Path.
             overwrite: Overwrite any existing files.
-            progress_bar: Show progress bar during download.
 
         Returns:
-            Path to downloaded file.
+            None.
 
         Raises:
-            planet.exceptions.APIException: On API error.
-        """
-        req = self._request(location, method='GET')
-
-        async with self._session.stream(req) as resp:
-            body = StreamingBody(resp)
-            dl_path = os.path.join(directory or '.', filename or body.name)
-            await body.write(dl_path,
-                             overwrite=overwrite,
-                             progress_bar=progress_bar)
-        return dl_path
+            OrderError
 
-    async def download_order(self,
-                             order_id: str,
-                             directory: str = None,
-                             overwrite: bool = False,
-                             progress_bar: bool = False) -> typing.List[str]:
-        """Download all assets in an order.
-
-        Parameters:
-            order_id: The ID of the order
-            directory: Write to given directory instead of current directory.
-            overwrite: Overwrite any existing files.
-            progress_bar: Show progress bar during download.
+        """
+        semaphore = asyncio.Semaphore(MAX_CONCURRENCY)
 
-        Returns:
-            Paths to downloaded files.
+        async with ClientSession() as session:
 
-        Raises:
-            planet.exceptions.APIException: On API error.
-        """
-        order = await self.get_order(order_id)
-        locations = order.locations
-        LOGGER.info(
-            f'downloading {len(locations)} assets from order {order_id}')
-        filenames = [
-            await self.download_asset(location,
-                                      directory=directory,
-                                      overwrite=overwrite,
-                                      progress_bar=progress_bar)
-            for location in locations
-        ]
```

Comment:
Since Python 3.6 comprehensions allow `async` in the expression but this list comprehension makes downloads one at a time, not concurrently.